### PR TITLE
feat: add Claude Agent SDK instrumentation

### PR DESCRIPTION
**Pull request by XinweiHe**
## Summary
- add TraceRoot-owned Claude Agent SDK instrumentation for query, LLM, subagent, and tool spans
- keep Claude Agent query spans structural while child anthropic.messages.create spans own model/token accounting
- wire manual Claude Agent SDK instrumentation through the TraceRoot wrapper and avoid divergent upstream auto instrumentation for now
- add regression coverage for span shape, usage grouping, tool ordering, and query cleanup edge cases

## Testing
- pnpm --filter @traceroot-ai/traceroot test -- claude-agent-sdk span-attributes openai-agents
- pnpm --filter @traceroot-ai/traceroot build

---
## Summary by cubic
Add TraceRoot’s Claude Agent SDK instrumentation to emit structured query, LLM, subagent, and tool spans with correct parenting and token/model attribution. Replaces upstream auto-instrumentation with a stable wrapper and adds tests for span shape, lifecycle, and iterator edge cases.

- New Features
  - Introduced `wireClaudeAgentSDKInstrumentation` to instrument `query` with TraceRoot’s wrapper.
  - LLM spans (`anthropic.messages.create`) own model and token counts (incl. cache read/creation); query span stays structural and holds prompt/session/summary data.
  - Smart parenting: tools attach to the active LLM turn or task/subagent; `Agent`/`Task`/`Bash` remain task siblings.
  - Subagent spans carry `agent_id`/`agent_type`, track lifecycle, and handle errors/cleanup. Groups assistant chunks by message ID and adjusts final usage from the result message.
  - Robust iterator lifecycle handling: spans close on construction errors and on double return/throw; tests cover ordering when hooks arrive before usage and not parenting under ended LLM spans.

- Refactors
  - Removed `@arizeai/openinference-instrumentation-claude-agent-sdk` from `instrumentation.ts` and now call the in-house `wireClaudeAgentSDKInstrumentation` when `instrumentModules.claudeAgentSDK` is provided.

<sup>Written for commit d2ee21489c1f69fddc0cffa0c2ec4e8f6c781e62. Summary will update on new commits. <a href="https://cubic.dev/pr/traceroot-ai/traceroot-ts/pull/97?utm_source=github">Review in cubic</a></sup>

## Screenshots

Before
Screenshot 2026-05-24 at 5 37 39 PM

After
Screenshot 2026-05-24 at 5 37 15 PM
Screenshot 2026-05-24 at 5 37 05 PM

### CODE REVIEW

File: packages/traceroot/src/claude-agent-sdk.ts [hunk 1]
```diff
@@ -0,0 +1,868 @@
...
+            role: message.message?.role ?? 'assistant',
+            content: message.message?.content,
+          }
+        : undefined,
+    )
+    .filter((message): message is { role: string; content: unknown } => message !== undefined);
+  if (output.length > 0) {
+    setJsonAttribute(span, OI_OUTPUT_VALUE, output);
+  }
+  span.setStatus({ code: SpanStatusCode.OK });
```
[the remark this answers is not in the document]
Comment (XinweiHe) [reply packages/traceroot/src/claude-agent-sdk.ts:587]: Got it, will handle this soon.
